### PR TITLE
Fix SQL printing of CastFunction target type

### LINKED ISSUE #16973: Unable to create a generated column involving a explicit cast from geo_point to array
### CrateDB version

5.9.2 and crate/crate:nightly-5.10.0-2024-11-12-00-02

### CrateDB setup information

Single node

### Problem description

Trying to dynamically generate shapes in generated columns using points stored as geo_points involves first casting the points to arrays, this seems to fail when the casting is explicit.

### Steps to Reproduce

```sql
SELECT [1,2]::GEO_POINT::ARRAY(DOUBLE);
--> casting GEO_POINT to ARRAY(DOUBLE) is OK;

CREATE TABLE test (a geo_point);
ALTER TABLE test ADD COLUMN b ARRAY(DOUBLE) GENERATED ALWAYS AS (a);
--> having a generated column with an implicit cast to ARRAY(DOUBLE) is ok

DROP TABLE test;
CREATE TABLE test (a geo_point);
ALTER TABLE test ADD COLUMN b ARRAY(DOUBLE) GENERATED ALWAYS AS (a::ARRAY(DOUBLE));
```

### Actual Result

```
SQLParseException[line 1:18: extraneous input 'precision_array' expecting ')']

io.crate.exceptions.SQLParseException: line 1:18: extraneous input 'precision_array' expecting ')'
        at io.crate.exceptions.SQLExceptions.esToCrateException(SQLExceptions.java:211)
        at io.crate.exceptions.SQLExceptions.prepareForClientTransmission(SQLExceptions.java:200)
        at io.crate.rest.action.SqlHttpHandler.sendResponse(SqlHttpHandler.java:176)
        at io.crate.rest.action.SqlHttpHandler.lambda$channelRead0$0(SqlHttpHandler.java:130)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:907)
        at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:885)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:554)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2238)
        at io.crate.rest.action.RestRowCountReceiver.fail(RestRowCountReceiver.java:84)
        at io.crate.session.RowConsumerToResultReceiver.accept(RowConsumerToResultReceiver.java:63)
        at io.crate.execution.support.OneRowActionListener.accept(OneRowActionListener.java:74)
        at io.crate.execution.support.OneRowActionListener.accept(OneRowActionListener.java:38)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:907)
        at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(CompletableFuture.java:885)
        at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:554)
        at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2238)
        at io.crate.action.FutureActionListener.onFailure(FutureActionListener.java:37)
        at org.elasticsearch.action.support.master.TransportMasterNodeAction$AsyncSingleAction.lambda$doStart$2(TransportMasterNodeAction.java:173)
        at org.elasticsearch.action.ActionListener$6.onFailure(ActionListener.java:276)
        at io.crate.execution.ddl.AbstractDDLTransportAction$1.onFailure(AbstractDDLTransportAction.java:89)
        at org.elasticsearch.cluster.service.MasterService$SafeClusterStateTaskListener.onFailure(MasterService.java:477)
        at org.elasticsearch.cluster.service.MasterService$TaskOutputs.notifyFailedTasks(MasterService.java:404)
        at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:176)
        at org.elasticsearch.cluster.service.MasterService.lambda$doStart$0(MasterService.java:121)
        at org.elasticsearch.cluster.service.TaskBatcher.runIfNotProcessed(TaskBatcher.java:170)
        at org.elasticsearch.cluster.service.BatchedTask.run(BatchedTask.java:70)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:231)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:194)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
        at java.base/java.lang.Thread.run(Thread.java:1575)
Caused by: io.crate.sql.parser.ParsingException: line 1:18: extraneous input 'precision_array' expecting ')'
        at io.crate.sql.parser.SqlParser$1.syntaxError(SqlParser.java:58)
        at org.antlr.v4.runtime.ProxyErrorListener.syntaxError(ProxyErrorListener.java:41)
        at org.antlr.v4.runtime.Parser.notifyErrorListeners(Parser.java:544)
        at io.crate.sql.parser.ExceptionErrorStrategy.reportUnwantedToken(ExceptionErrorStrategy.java:53)
        at org.antlr.v4.runtime.DefaultErrorStrategy.singleTokenDeletion(DefaultErrorStrategy.java:548)
        at org.antlr.v4.runtime.DefaultErrorStrategy.recoverInline(DefaultErrorStrategy.java:467)
        at org.antlr.v4.runtime.Parser.match(Parser.java:208)
        at io.crate.sql.parser.antlr.SqlBaseParser.explicitFunction(SqlBaseParser.java:12031)
        at io.crate.sql.parser.antlr.SqlBaseParser.primaryExpression(SqlBaseParser.java:11006)
        at io.crate.sql.parser.antlr.SqlBaseParser.valueExpression(SqlBaseParser.java:10444)
        at io.crate.sql.parser.antlr.SqlBaseParser.predicated(SqlBaseParser.java:9714)
        at io.crate.sql.parser.antlr.SqlBaseParser.booleanExpression(SqlBaseParser.java:9563)
        at io.crate.sql.parser.antlr.SqlBaseParser.expr(SqlBaseParser.java:9182)
        at io.crate.sql.parser.antlr.SqlBaseParser.singleExpression(SqlBaseParser.java:488)
        at io.crate.sql.parser.SqlParser.invokeParser(SqlParser.java:128)
        at io.crate.sql.parser.SqlParser.generateExpression(SqlParser.java:103)
        at io.crate.sql.parser.SqlParser.createExpression(SqlParser.java:99)
        at io.crate.metadata.doc.DocTableInfoFactory.create(DocTableInfoFactory.java:294)
        at io.crate.execution.ddl.tables.AlterTableTask.execute(AlterTableTask.java:60)
        at io.crate.metadata.cluster.DDLClusterStateTaskExecutor.execute(DDLClusterStateTaskExecutor.java:38)
        at org.elasticsearch.cluster.service.MasterService.executeTasks(MasterService.java:665)
        at org.elasticsearch.cluster.service.MasterService.calculateTaskOutputs(MasterService.java:280)
        at org.elasticsearch.cluster.service.MasterService.runTasks(MasterService.java:175)
        ... 8 more
```

### Expected Result

```
ALTER OK
```

### PR DESCRIPTION
**Pull request by seut**
To print a data type as valid SQL including all optional type parameters, the `DataType.toColumnType()` + SQLFormatter must be used.

Fixes #16973.

### CODE REVIEW

File: docs/appendices/release-notes/5.9.3.rst [hunk 1]
```diff
@@ -115,3 +115,12 @@ Fixes
 
       CREATE FOREIGN TABLE t (a INT) SERVER s;
       SELECT * FROM (SELECT id as some_alias FROM t) tt WHERE tt.some_alias = 1;
+
+- Fixed an issue that may caused the use of explicit casts inside a generated
```
Comment (matriv) [inline docs/appendices/release-notes/5.9.3.rst:119]:
```suggestion
- Fixed an issue that may cause the use of explicit casts inside a generated
```

**Review by matriv — APPROVED** [review]
thx!

**Comment by matriv** [comment]
ah, sorry, now I saw there are test failures, so please ping again once fixed.

**Comment by seut** [comment]
> ah, sorry, now I saw there are test failures, so please ping again once fixed.

fixed

**Review by matriv — APPROVED** [review]
thx!

**Pull request merged**